inst/shiny/modules/tlg_plot.R

```diff
@@ -0,0 +1,229 @@
+              plot module.
+            "),
+            tags$p("Leaving a widget empty will allow default behaviour of the plotting function."),
+            tags$p(
+              "In text fields, you can reference values / columns in the dataset by using
+              the dollar sign (", tags$b("$"), ") and providing column name, e.g. ",
+              tags$b("$DOSEU"), "."
+            ),
+            tags$p("
+              You can also reference ", tags$i("label"), " attribute of any column by prefacing the
+              column name by exclamation mark (", tags$b("!"), "), e.g. ", tags$b("!DOSEU"), ".
+            ")
+          ),
+          style = "unite",
+          right = TRUE,
+          icon = icon("question"),
+          status = "primary"
+        ),
+        actionButton(
+          inputId = ns("reset_widgets"),
+          label = "Reset to defaults"
+        ),
+        uiOutput(ns("options"), class = "plot-options-container")
+      )
+    ),
+    div(
+      class = "plot-widgets-container",
+      div(
+        align = "left",
+        tags$span(
+          class = "inline-select-input",
+          style = "margin-right: 5em;",
+          tags$span("Plots per page:"),
+          selectInput(
+            ns("plots_per_page"),
+            "",
+            choices = c("All", 1, 2, 4, 6, 8, 10)
+          )
+        ),
+        shinyjs::disabled(actionButton(ns("previous_page"), "Previous Page", class = "btn-page"))
+      ),
+      div(
+        align = "center",
+        tags$span(
+          class = "inline-select-input",
+          tags$span("Page "),
+          selectInput(
+            inputId = ns("select_page"),
+            label = "",
+            choices = ""
+          ),
+          tags$span(" out of "),
+          uiOutput(ns("page_number"), inline = TRUE),
+        )
+      ),
+      div(align = "right", actionButton(ns("next_page"), "Next Page", class = "btn-page"))
+    ),
+    shinycssloaders::withSpinner(
+      uiOutput(ns("plots"))
+    )
+  )
+}
+
+tlg_plot_server <- function(id, render_plot, options = NULL) {
+  moduleServer(id, function(input, output, session) {
+    data <- session$userData$data
+
+    current_page <- reactiveVal(1)
+
+    #' updating current page based on user input
+    observeEvent(input$next_page, current_page(current_page() + 1))
+    observeEvent(input$previous_page, current_page(current_page() - 1))
+    observeEvent(input$select_page, {
+      if (input$select_page == "") return(NULL)
+      current_page(as.numeric(input$select_page))
+    })
+
+    #' hold reactive information about the page layout
+    num_pages <- reactive({
+      req(plot_list(), plots_per_page())
+      ceiling(length(plot_list()) / plots_per_page())
+    })
+
+    plots_per_page <- reactive({
+      if (is.null(input$plots_per_page)) return(NULL)
+      if (is.null(plot_list())) return(NULL)
+      if (input$plots_per_page == "All") {
+        isolate(length(plot_list()))
```

Comment:
nitpick: isolate here has no effect, because `plot_list()` is not isolated 2 lines above